src/cargo/core/compiler/standard_lib.rs

```diff
@@ -153,6 +154,7 @@ pub fn resolve_std<'cfg>(
         &specs,
         HasDevUnits::No,
         crate::core::resolver::features::ForceAllTargets::No,
+        max_rust_version,
```

Comment:
This is one case I'm not entirely sure of but since its another unstable feature, I figure doing a best effort isn't the end of the world

Reply:
I'm assuming this is build-std and we do a separate resolve pass for that.  What should we consider `max_rust_version` to be in this case?